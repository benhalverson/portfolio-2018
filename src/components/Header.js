import React from 'react';
import { NavLink } from 'react-router-dom';
import { Nav, NavItem } from 'reactstrap';

const Header = () => (
  <header>
    <h1>Portfolio header</h1>
    <nav>
      <NavLink to="/" activeClassName="is-active" exact={true}>
        About
      </NavLink>

      <NavLink to="/portfolio" activeClassName="is-active">
        Portfolio
      </NavLink>
      
      <NavLink to="/blog" activeClassName="is-active">
        Blog
      </NavLink>
        
      <NavLink to="/contact" activeClassName="is-active">
        Contact
      </NavLink>
    </nav>
  </header>
);

export default Header;