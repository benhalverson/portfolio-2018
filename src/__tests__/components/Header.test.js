import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';
test('it should render the Header', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});
