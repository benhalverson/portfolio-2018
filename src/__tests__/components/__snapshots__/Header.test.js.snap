// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the Header 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Portfolio header
</h1>,
        <nav>
          <NavLink
                    activeClassName="is-active"
                    aria-current="page"
                    exact={true}
                    to="/"
          >
                    About
          </NavLink>
          <NavLink
                    activeClassName="is-active"
                    aria-current="page"
                    to="/portfolio"
          >
                    Portfolio
          </NavLink>
          <NavLink
                    activeClassName="is-active"
                    aria-current="page"
                    to="/blog"
          >
                    Blog
          </NavLink>
          <NavLink
                    activeClassName="is-active"
                    aria-current="page"
                    to="/contact"
          >
                    Contact
          </NavLink>
</nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Portfolio header",
        },
        "ref": null,
        "rendered": "Portfolio header",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="is-active"
              aria-current="page"
              exact={true}
              to="/"
>
              About
</NavLink>,
            <NavLink
              activeClassName="is-active"
              aria-current="page"
              to="/portfolio"
>
              Portfolio
</NavLink>,
            <NavLink
              activeClassName="is-active"
              aria-current="page"
              to="/blog"
>
              Blog
</NavLink>,
            <NavLink
              activeClassName="is-active"
              aria-current="page"
              to="/contact"
>
              Contact
</NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "is-active",
              "aria-current": "page",
              "children": "About",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "About",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "is-active",
              "aria-current": "page",
              "children": "Portfolio",
              "to": "/portfolio",
            },
            "ref": null,
            "rendered": "Portfolio",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "is-active",
              "aria-current": "page",
              "children": "Blog",
              "to": "/blog",
            },
            "ref": null,
            "rendered": "Blog",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "is-active",
              "aria-current": "page",
              "children": "Contact",
              "to": "/contact",
            },
            "ref": null,
            "rendered": "Contact",
            "type": [Function],
          },
        ],
        "type": "nav",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Portfolio header
</h1>,
          <nav>
            <NavLink
                        activeClassName="is-active"
                        aria-current="page"
                        exact={true}
                        to="/"
            >
                        About
            </NavLink>
            <NavLink
                        activeClassName="is-active"
                        aria-current="page"
                        to="/portfolio"
            >
                        Portfolio
            </NavLink>
            <NavLink
                        activeClassName="is-active"
                        aria-current="page"
                        to="/blog"
            >
                        Blog
            </NavLink>
            <NavLink
                        activeClassName="is-active"
                        aria-current="page"
                        to="/contact"
            >
                        Contact
            </NavLink>
</nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Portfolio header",
          },
          "ref": null,
          "rendered": "Portfolio header",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="is-active"
                aria-current="page"
                exact={true}
                to="/"
>
                About
</NavLink>,
              <NavLink
                activeClassName="is-active"
                aria-current="page"
                to="/portfolio"
>
                Portfolio
</NavLink>,
              <NavLink
                activeClassName="is-active"
                aria-current="page"
                to="/blog"
>
                Blog
</NavLink>,
              <NavLink
                activeClassName="is-active"
                aria-current="page"
                to="/contact"
>
                Contact
</NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "is-active",
                "aria-current": "page",
                "children": "About",
                "exact": true,
                "to": "/",
              },
              "ref": null,
              "rendered": "About",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "is-active",
                "aria-current": "page",
                "children": "Portfolio",
                "to": "/portfolio",
              },
              "ref": null,
              "rendered": "Portfolio",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "is-active",
                "aria-current": "page",
                "children": "Blog",
                "to": "/blog",
              },
              "ref": null,
              "rendered": "Blog",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "is-active",
                "aria-current": "page",
                "children": "Contact",
                "to": "/contact",
              },
              "ref": null,
              "rendered": "Contact",
              "type": [Function],
            },
          ],
          "type": "nav",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
